projects/miopen/test/gemm.hpp

```
@@ -26,44 +26,95 @@
-void gemm(std::size_t n, std::size_t m, std::size_t k, AF a, BF b, CF c)
+/*
+    A and B rows and cols should be passed as default values (NxM, MxK), independently of
+   a_transponse/b_transpose flag value
+    C rows and cols should have correct values based on a_transponse/b_transpose values
+    A, B, C strides should have corret values based on a_transponse/b_transpose values
+*/
+template <typename Dtype>
+void gemm_cpu(const Dtype* a_ptr,
+              const size_t a_cols,
+              const size_t a_rows,
+              const size_t a_stride,
+              const bool a_transpose,
+              const Dtype* b_ptr,
+              const size_t b_cols,
+              const size_t b_rows,
+              const size_t b_stride,
+              const bool b_transpose,
+              Dtype* c_ptr,
+              const size_t c_cols,
+              const size_t c_rows,
+              const size_t c_stride,
+              double alpha = 1.0,
+              double beta  = 1.0)
 {
-    auto inner_loop = [&](int i, int j) {
-        double x = 0.0;
-        miopen::ford(k)([&](int kk) { x += a(i, kk) * b(kk, j); });
-        c(i, j, x);
+    if((!a_transpose && !b_transpose &&
+        ((a_cols != b_rows) || (a_rows != c_rows) || (b_cols != c_cols))) ||
+       (a_transpose && b_transpose &&
+        ((a_rows != b_cols) || (a_cols != c_rows) || (b_rows != c_cols))) ||
+       (a_transpose && !b_transpose &&
+        ((a_rows != b_rows) || (a_cols != c_rows) || (b_cols != c_cols))) ||
+       (!a_transpose && b_transpose &&
+        ((a_cols != b_cols) || (a_rows != c_rows) || (b_rows != c_cols))))
+    {
+
+        std::cout << "MM_CPU_ERROR. Incompatible matrix size: " << std::endl
+                  << "A: " << a_rows << "x" << a_cols << " transpose: " << a_transpose << std::endl
+                  << "B: " << b_rows << "x" << b_cols << " transpose: " << b_transpose << std::endl
+                  << "C: " << c_rows << "x" << c_cols << std::endl;
+        return;
```

This function silently returns on dimension mismatch without any indication of failure, its better recommended to:

```
    throw std::runtime_error("Incompatible matrix dimensions in gemm_cpu");

```

---

Will use `MIOPEN_THROW`